packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -190,6 +563,181 @@ export const logExperimentUpdated = async ({
+
+  await ExperimentModel.updateMany(query, {
+    $pull: { tags: tag },
+  });
+
+  previousExperiments.forEach((previous) => {
+    const current = cloneDeep(previous);
+    current.tags = current.tags.filter((t) => t != tag);
+
+    logExperimentUpdated({
+      organization,
+      previous,
+      current,
+    });
+  });
+};
+
+export async function getExperimentsByOrganization(
+  organization: string,
+  project?: string
+) {
+  const query: FilterQuery<ExperimentDocument> = {
+    organization,
+  };
+
+  if (project) {
+    query.project = project;
+  }
+
+  const experiments = await ExperimentModel.find(query);
+
+  return experiments.map(toInterface);
+}
+
+export async function removeMetricFromExperiments(
+  metricId: string,
+  organization: OrganizationInterface
+) {
+  const oldExperiments: Record<
+    string,
+    {
+      previous: ExperimentInterface | null;
+      current: ExperimentInterface | null;
+    }
+  > = {};
+
+  const orgId = organization.id;
+
+  const metricQuery = { organization: orgId, metrics: metricId };
+  const guardRailsQuery = { organization: orgId, guardrails: metricId };
+  const activationMetricQuery = {
+    organization: orgId,
+    activationMetric: metricId,
+  };
+  const docsToTrackChanges = await ExperimentModel.find({
+    $or: [metricQuery, guardRailsQuery, activationMetricQuery],
+  });
+  docsToTrackChanges.forEach((experiment: ExperimentDocument) => {
+    if (!oldExperiments[experiment.id]) {
+      oldExperiments[experiment.id] = {
+        previous: experiment,
+        current: null,
+      };
+    }
+  });
+
+  // Remove from metrics
+  await ExperimentModel.updateMany(metricQuery, {
+    $pull: { metrics: metricId },
+  });
+
+  // Remove from guardrails
+  await ExperimentModel.updateMany(guardRailsQuery, {
+    $pull: { guardrails: metricId },
+  });
+
+  // Remove from activationMetric
+  await ExperimentModel.updateMany(activationMetricQuery, {
+    $set: { activationMetric: "" },
+  });
+
+  const ids = Object.keys(oldExperiments);
+  const updatedExperiments = await ExperimentModel.find({ id: { $in: ids } });
+  // Populate updated experiments
+  updatedExperiments.forEach((experiment) => {
+    const changeSet = oldExperiments[experiment.id];
+    if (changeSet) {
+      changeSet.current = experiment;
+    }
+  });
+
+  // Log all the changes
+  each(oldExperiments, async (changeSet) => {
+    const { previous, current } = changeSet;
+    if (current && previous) {
+      await logExperimentUpdated({
+        organization,
+        current,
+        previous,
+      });
+    }
+  });
+}
+
+export async function removeProjectFromExperiments(
+  project: string,
+  organization: OrganizationInterface
+) {
+  const query = { organization: organization.id, project };
+  const previousExperiments = await ExperimentModel.find(query);
+
+  await ExperimentModel.updateMany(query, { $set: { project: "" } });
+
+  previousExperiments.forEach((previous) => {
```

Comment:
Need to use toInterface here

Reply:
Updated this to:

```
export async function removeProjectFromExperiments(
  project: string,
  organization: OrganizationInterface
) {
  const query = { organization: organization.id, project };
  const previousExperiments = await ExperimentModel.find(query);

  await ExperimentModel.updateMany(query, { $set: { project: "" } });

  const previousExperimentsInterface = previousExperiments.map(toInterface);

  previousExperimentsInterface.forEach((previous) => {
    const current = cloneDeep(previous);
    current.project = "";

    logExperimentUpdated({
      organization,
      previous,
      current,
    });
  });
}
```

I feel like there is a better pattern than creating the `previousExperimentsInterface` array only to loop through it.

The alternative of chaining previousExperiments.map(toInterface).forEach((previous) =>.....` feels very verbose, and a bit hard to follow, though.

Any suggestions?